conans/model/build_info.py

```diff
@@ -35,6 +35,56 @@ def __copy__(self):
         return the_copy
 
 
+class BuildModulesDict(dict):
+    """
+    A dictionary with append and extend for cmake build modules to keep it backwards compatible
+    with the list interface
+    """
+
+    def __getitem__(self, key):
+        if key not in self.keys():
+            super(BuildModulesDict, self).__setitem__(key, list())
+        return super(BuildModulesDict, self).__getitem__(key)
+
+    def append(self, item):
+        if item.endswith(".cmake"):
+            self["cmake"].append(item)
+            self["cmake_multi"].append(item)
+            self["cmake_find_package"].append(item)
+            self["cmake_find_package_multi"].append(item)
```

Comment:
What if I'm already adding files with other extensions? Now they will be lost...

```python
self.cpp_info.build_modules.append("mything.pc")
```

Maybe here it is breaking and we can break (no idea if we already have any recipe with build-modules for other generators, at least in `conan-center-index`), we can raise and say that build-modules at global level are not allowed. wdyt?

Reply:
Maybe better a warning (error using conan_v2_mode)